tests/snapshot/snapshot.go

```diff
@@ -0,0 +1,401 @@
+	require.NoError(t, err)
+
+	steveclient, err := client.Steve.ProxyDownstream(clusterID)
+	require.NoError(t, err)
+
+	localClusterID, err := clusters.GetClusterIDByName(client, localClusterName)
+	require.NoError(t, err)
+
+	var isRKE1 bool
+
+	clusterObject, _, _ := clusters.GetProvisioningClusterByName(client, clusterName, namespace)
+	if clusterObject == nil {
+		_, err := client.Management.Cluster.ByID(clusterID)
+		require.NoError(t, err)
+
+		isRKE1 = true
+	} else {
+		isRKE1 = false
+	}
+
+	containerTemplate := workloads.NewContainer(containerName, containerImage, corev1.PullAlways, []corev1.VolumeMount{}, []corev1.EnvFromSource{}, nil, nil, nil)
+	podTemplate := workloads.NewPodTemplate([]corev1.Container{containerTemplate}, []corev1.Volume{}, []corev1.LocalObjectReference{}, nil)
+	deployment := workloads.NewDeploymentTemplate(initialWorkloadName, defaultNamespace, podTemplate, isCattleLabeled, nil)
+
+	service := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAppendName + initialWorkloadName,
+			Namespace: defaultNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{
+					Name: port,
+					Port: 80,
+				},
+			},
+			Selector: deployment.Spec.Template.Labels,
+		},
+	}
+
+	deploymentResp, serviceResp, err := workloads.CreateDeploymentWithService(steveclient, initialWorkloadName, deployment, service)
+	require.NoError(t, err)
+
+	err = workloads.VerifyDeployment(steveclient, deploymentResp)
+	require.NoError(t, err)
+	require.Equal(t, initialWorkloadName, deploymentResp.ObjectMeta.Name)
+
+	path := ingresses.NewIngressPathTemplate(networking.PathTypeExact, ingressPath, serviceAppendName+initialWorkloadName, 80)
+	ingressTemplate := ingresses.NewIngressTemplate(initialIngressName, defaultNamespace, "", []networking.HTTPIngressPath{path})
+
+	ingressResp, err := ingresses.CreateIngress(steveclient, initialIngressName, ingressTemplate)
+	require.NoError(t, err)
+	require.Equal(t, initialIngressName, ingressResp.ObjectMeta.Name)
+
+	if isRKE1 {
+		snapshotRestoreRKE1(t, client, podTemplate, deployment, clusterName, clusterID, localClusterID, clusterConfig, isRKE1, terraformOptions)
+	} else {
+		snapshotRestoreRKE2K3S(t, client, podTemplate, deployment, clusterName, clusterID, localClusterID, clusterConfig, isRKE1, terraformOptions)
+	}
+
+	logrus.Infof("Deleting created workloads...")
+	err = steveclient.SteveType(DeploymentSteveType).Delete(deploymentResp)
+	require.NoError(t, err)
+
+	err = steveclient.SteveType(ServiceType).Delete(serviceResp)
+	require.NoError(t, err)
+
+	err = steveclient.SteveType(IngressSteveType).Delete(ingressResp)
+	require.NoError(t, err)
+}
+
+func snapshotRestoreRKE1(t *testing.T, client *rancher.Client, podTemplate corev1.PodTemplateSpec, deployment *v1.Deployment, clusterName, clusterID, localClusterID string, clusterConfig *config.TerratestConfig, isRKE1 bool, terraformOptions *terraform.Options) {
```

Comment:
super nit; might make more sense to break this out into `snapshotRKE1` and `restoreRKE1` and equivalent for rke2/k3s so that this is more modular for future test cases.

Reply:
Yeah the majority of this is just copy/paste from `rancher/shepherd`. If it is alright with you, let's make the change there first and update here? Just for parity so it's not confusing for the time being.